packages/api/internal/template-manager/logs/template_manager_provider.go

```diff
@@ -0,0 +1,54 @@
+package logs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.uber.org/zap"
+	"google.golang.org/grpc/metadata"
+
+	template_manager "github.com/e2b-dev/infra/packages/api/internal/template-manager"
+	templatemanagergrpc "github.com/e2b-dev/infra/packages/shared/pkg/grpc/template-manager"
+	"github.com/e2b-dev/infra/packages/shared/pkg/logger"
+	"github.com/e2b-dev/infra/packages/shared/pkg/telemetry"
+)
+
+type TemplateManagerProvider struct {
+	TemplateManager *template_manager.TemplateManager
+}
+
+func (t *TemplateManagerProvider) GetLogs(ctx context.Context, templateID string, buildUUID uuid.UUID, clusterID *uuid.UUID, clusterNodeID *string, offset *int32) ([]string, error) {
+	grpc, _, err := t.TemplateManager.GetBuilderClient(clusterID, clusterNodeID, false)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get builder edgeHttpClient: %w", err)
+	}
+
+	logs := make([]string, 0)
+
+	reqCtx := metadata.NewOutgoingContext(ctx, grpc.Metadata)
+	res, err := grpc.Client.Template.TemplateBuildStatus(
+		reqCtx, &templatemanagergrpc.TemplateStatusRequest{
```

Comment:
do you want to return all logs (no offset)?

Reply:
That you will send all logs on each status call, basically asking if it isn't better to propagate the offset